lab3/main2.c

```diff
@@ -0,0 +1,286 @@
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <math.h>
+
+
+#define EPSILON 0.000000001
+
+typedef enum {
+    SUCCESS,
+    MEMORY_ERROR,
+} ret_type_t;
+
+typedef struct{
+    int N;
+    double* coords;
+} VVector;
+
+int VVector_delete(VVector* v) {
+    if(v == NULL){
+        return 0;
+    }
+    else if(v->coords == NULL){
+        free(v);
+    }
+    free(v->coords);
+    free(v);
+    return 1;
+}
+
+VVector* Vvector_copy(VVector* v) {
+    VVector* result = malloc(sizeof(VVector));
+    double* coord = malloc(sizeof(double)*  v->N);
+    for(int i = 0; i < v->N; i++) {
+        coord[i] = v->coords[i];
+    }
+    result->coords = coord;
+    result->N = v->N;
+    return result;
+}
+
+VVector* Vvector_init(int n, ...) {
+    va_list list;
+    double* coords = malloc(sizeof(double)*  n);
+    VVector* result = malloc(sizeof(VVector));
+    if (coords == NULL) {
+        return NULL;
+    }
+
+    if (!result) {
+        free(coords);
+        return NULL;
+    }
+
+    va_start(list, n);
+    for (int i = 0; i < n; i++) {
+        coords[i] = va_arg(list, double);
+    }
+    va_end(list);
+    result->coords = coords;
+    result->N = n;
+    return result;
+}
+
+typedef double (*callback)(const VVector* , const void* );
+
+typedef struct{
+    VVector** data;
+    int len;
+    int _size;
+} Vector;
+
+Vector* Vector_init(int size) {
+    Vector* v = malloc(sizeof(Vector));
+    if (v == NULL) {
+        return NULL;
+    }
+
+    v->data = malloc(sizeof(VVector* )*  size);
+    if (v->data == NULL) {
+        free(v);
+        return NULL;
+    }
+    v->_size = size;
+    v->len = 0;
+    return v;
+}
+
+int Vector_add(Vector* v, VVector* value) {
+    if (v->len + 1 >= v->_size) {
+        VVector** newdata = malloc(v->_size*  sizeof(VVector* )*  2);
+        for (int i = 0; i < v->len; i++) {
+            newdata[i] = v->data[i];
+        }
+        free(v->data);
+        v->data = newdata;
+        v->_size *= 2;
+    }
+    v->data[v->len] = value;
+    v->len++;
+    return 1;
+}
+
+int Vector_pop(Vector* v) {
+    if(v == NULL){
+        return 0;
+    }
+    else if(v->len == 0 || v->data == NULL ){
+        return 0;
+    }
+    v->len--;
+    free(v->data[v->len]->coords);
+    free(v->data[v->len]);
+    return 1;
+}
+
+int Vector_delete(Vector* v) {
+    if(v == NULL){
+        return 0;
+    }
+    else if(v->data == NULL || v->len == 0){
+        free(v);
+        return 0;
+    }
+    for (int i = 0; i < v->len; i++) {
+        if(v->data[i]->coords != NULL){
+            free(v->data[i]->coords);
+        }
+        if(v->data[i] != NULL){
+            free(v->data[i]);
+        }
+    }
+
+    if(v->data != NULL){
+        free(v->data);
+    }
+    if(v != NULL){
+        free(v);
+    }
+    return 1;
+};
+
+double normMax(const VVector* v, const void* arg) {
+
```

Comment:
Можно было добавить такую строку и ошибка бы ушла
```suggestion
    (void)arg;
```